### PR TITLE
add zero-diff check to status controller sync method

### PR DESCRIPTION
**Pull request by juanvallejo**
cc @deads2k

### CODE REVIEW

File: pkg/operator/status/controller.go [hunk 1]
```diff
@@ -122,6 +125,10 @@ func (c StatusSyncer) sync() error {
 	}
 	unstructured.SetNestedSlice(operatorConfig.Object, conditions, "status", "conditions")
 
+	if reflect.DeepEqual(originalConfig, operatorConfig) {
```
Comment (deads2k) [inline pkg/operator/status/controller.go]: try `equalities.Semantic.DeepEqual` (apimachinery).

File: pkg/operator/status/controller.go [hunk 2]
```diff
@@ -79,6 +80,8 @@ func (c StatusSyncer) sync() error {
 	if err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
+	originalConfig := operatorConfig.DeepCopy()
```
Comment (deads2k) [inline pkg/operator/status/controller.go]: realizing I started this pattern and we actually want the copy to be the one we modify.

**Comment by juanvallejo** [comment]
@deads2k thanks, review comments addressed

**Comment by deads2k** [comment]
link to a fake bump pull you've done and tested loclaly that makes sure that this works.

once that's confirmed, this lgtm

**Comment by deads2k** [comment]
travis runs go test. merging

**Pull request merged**